File: formasionAPI/src/main/java/fr/formation/formasion/model/Module.java
package fr.formation.formasion.model;

import java.time.LocalDate;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Module {

	@Id
	private String id;

	private String libelle;

	private LocalDate dateDebut;

	private LocalDate dateFin;

	@ManyToOne
	@JoinColumn(name = "personne_id")
	@JsonIgnoreProperties("modules")
	private Personne formateur;

	@ManyToOne
	@JoinColumn(name = "filiere_id")
	@JsonIgnoreProperties("modules")
	private Filiere filiere;

}

File: gestion-formasion/src/app/models/module.ts
import { Filiere } from "./filiere";
import { Personne } from "./personne";

export class Module {
    id: number;
    libelle: string;
    dateDebut: Date;
    dateFin: Date;
    formateur: Personne;
    filiere: Filiere;

    constructor(id: number, libelle: string, dateDebut: Date, dateFin: Date, formateur: Personne, filiere: Filiere) {
        this.id = id;
        this.libelle = libelle;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.formateur = formateur;
        this.filiere = filiere;
    }
}